### DISCUSSION TITLE
Ability to use Map in iframe

### DISCUSSION THREAD

**Discussion by tazo90** [maplibre/maplibre-gl-js#1588] [Q&A]
The map is not able to run inside iframe using react portal.
App crashes with an error: `Invalid type: 'container' must be a String or HTMLElement`
Related to: [#8036](https://github.com/mapbox/mapbox-gl-js/issues/8036)

**maplibre-gl-js version**: 2.4.0

**source**: [map.ts#L433](https://github.com/maplibre/maplibre-gl-js/blob/main/src/ui/map.ts#L433)

**browser**: chrome / firefox

### Steps to Trigger Behavior

render maplibre inside iframe 
 1. create a iframe
 2. use React.createPortal to render the map component
 3. render map inside iframe

### Link to Demonstration

https://codesandbox.io/s/render-in-iframe-2-forked-ut3zoq

### Expected Behavior

App should not crash

### Actual Behavior

App crashes with following error message: `Invalid type: 'container' must be a String or HTMLElement.`

**Comment by HarelM** [comment]
Not related to this repo. I know for a fact it can.
